server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/rfc/ResponseObjectTest.scala

```diff
@@ -0,0 +1,189 @@
+            |      "arg1": "arg1data",
+            |      "arg2": "arg2data"
+            |    }, "c1" ]
+            |""".stripMargin)})
+      Json.fromJson[Invocation](invocationJsValue) must be(JsSuccess(Invocation(invocationJsValue)))
+    }
+
+    "failed with wrong value type" in {
+      val invocationJsValue: JsValue = JsBoolean(true)
+      Json.fromJson[Invocation](invocationJsValue) must not be (JsSuccess(Invocation(Json.arr({
+        Json.parse(
+          """[ "Core/echo", {
+            |      "arg1": "arg1data",
+            |      "arg2": "arg2data"
+            |    }, "c1" ]
+            |""".stripMargin)}))))
+    }
+  }
+
+  "Serialize Invocation" must {
+    "succeed " in {
+      val invocation: Invocation = Invocation(Json.arr({
+        Json.parse(
+          """[ "Core/echo", {
+            |      "arg1": "arg1data",
+            |      "arg2": "arg2data"
+            |    }, "c1" ]
+            |""".stripMargin)}))
+      val expectedInvocationJsArray: JsArray = Json.arr({
+        Json.parse(
+          """[ "Core/echo", {
+            |      "arg1": "arg1data",
+            |      "arg2": "arg2data"
+            |    }, "c1" ]
+            |""".stripMargin)})
+
+      Json.toJson[Invocation](invocation) must be(expectedInvocationJsArray)
+    }
+  }
+
+  "Deserialize ResponseObject" must {
+    "succeed " in {
+      ResponseObject.deserialize(
+        """
+          |{
+          |  "methodResponses": [
+          |    [ "Core/echo", {
+          |      "arg1": "arg1data",
+          |      "arg2": "arg2data"
+          |    }, "c1" ]
+          |  ],
+          |  "sessionState": "75128aab4b1b"
+          |}
+          |""".stripMargin) must be(
+        ResponseObject.ResponseObject(
+          sessionState = ResponseObject.SessionState("75128aab4b1b"),
+          methodResponses = Seq(Invocation(Json.parse(
+            """[ "Core/echo", {
+              |      "arg1": "arg1data",
+              |      "arg2": "arg2data"
+              |    }, "c1" ]
+              |""".stripMargin).as[JsArray]))))
+    }
+
+    "succeed with many Capability, methodCalls" in {
+      ResponseObject.deserialize(
+        """
+          |{
+          |  "sessionState": "75128aab4b1b",
+          |  "methodResponses": [
+          |    [ "Core/echo", {
+          |      "arg1": "arg1data",
+          |      "arg2": "arg2data"
+          |    }, "c1" ],
+          |    [ "Core/echo2", {
+          |      "arg3": "arg3data",
+          |      "arg4": "arg4data"
+          |    }, "c2" ]
+          |  ]
+          |}
+          |""".stripMargin) must be(
+        ResponseObject.ResponseObject(
+          sessionState = ResponseObject.SessionState("75128aab4b1b"),
+          methodResponses = Seq(Invocation(Json.parse(
+            """[ "Core/echo", {
+              |      "arg1": "arg1data",
+              |      "arg2": "arg2data"
+              |    }, "c1" ]
+              |""".stripMargin).as[JsArray]), Invocation(Json.parse(
```

Comment:
I'm against the use of `Json.parse(` in the expected object: I can't check what the actual format is here...